File: token/token.go
package token

type TokenType string

type Token struct {
	Type    TokenType
	Literal string
}

const (
	ILLEGAL   = "ILLEGAL"
	EOF       = "EOF"
	IDENT     = "IDENT"
	INT       = "INT"
	PLUS      = "+"
	ASSIGN    = "="
	COMMA     = ","
	SEMICOLON = ";"
	LPAREN    = "("
	RPAREN    = ")"
	LBRACE    = "{"
	RBRACE    = "}"
	FUNCTION  = "FUNCTION"
	LET       = "LET"
	MINUS     = "-"
	BANG      = "!"
	ASTERISK  = "/"
	SLASH     = "/"
	LT        = "<"
	GT        = ">"
	TRUE      = "TRUE"
	FALSE     = "FALSE"
	IF        = "IF"
	ELSE      = "ELSE"
	RETURN    = "RETURN"
	EQ        = "=="
	NEQ       = "!="
)

var keywords = map[string]TokenType{
	"fn":     FUNCTION,
	"let":    LET,
	"true":   TRUE,
	"false":  FALSE,
	"if":     IF,
	"else":   ELSE,
	"return": RETURN,
}

func LookupIdent(ident string) TokenType {
	if tok, ok := keywords[ident]; ok {
		return tok
	}
	return IDENT
}

File: src/token/token.ts
type TokenType = string;

type Token = {
    Type: TokenType,
    Literal: string,
}

const ILLEGAL = "ILLEGAL";
const EOF = "EOF";

// Identifiers + literals
const IDENT = "IDENT"; // add, foobar, x, y, ...
const INT = "INT"; // 1343456

// Operators
const ASSIGN = "=";
const PLUS = "+";

// Delimiters
const COMMA = ",";
const SEMICOLON = ";";

const LPAREN = "(";
const RPAREN = ")";
const LBRACE = "{";
const RBRACE = "}";

// Keywords
const FUNCTION = "FUNCTION";
const LET = "LET";

export {
    ILLEGAL,
    EOF,

    IDENT,
    INT,
    
    ASSIGN,
    PLUS,

    COMMA,
    SEMICOLON,

    LPAREN,
    RPAREN,
    LBRACE,
    RBRACE,

    FUNCTION,
    LET,
}

export type {
    TokenType,
    Token,
}